test/unit/proc_macro/proc_macro_test.bzl

```diff
@@ -0,0 +1,100 @@
+"""Unittest to verify proc-macro targets"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest")
+load("//rust:defs.bzl", "rust_proc_macro", "rust_test")
+load("//test/unit:common.bzl", "assert_action_mnemonic", "assert_list_contains_adjacent_elements", "assert_list_contains_adjacent_elements_not")
+
+def _proc_macro_test_targets(edition):
+    """Define a set of `rust_proc_macro` targets for testing
+
+    Args:
+        edition (str): The rust edition to use for the new targets
+    """
+    rust_proc_macro(
+        name = "proc_macro_lib_{}".format(edition),
+        srcs = [
+            "proc_macro_{}.rs".format(edition),
+        ],
+        edition = edition,
+    )
+
+    rust_test(
+        name = "proc_macro_lib_{}_unittest".format(edition),
+        crate = ":proc_macro_lib_{}".format(edition),
+        edition = edition,
+    )
+
+    rust_test(
+        name = "proc_macro_lib_{}_integration_test".format(edition),
+        srcs = ["proc_macro_{}_test.rs".format(edition)],
+        edition = edition,
+        proc_macro_deps = [":proc_macro_lib_{}".format(edition)],
+    )
+
+def _proc_macro_2015_no_extern_flag_impl(ctx):
```

Comment:
I think we can be more explicit here. Please extract a function with the body of the unittests (besides asserting presence of flags they are identical).

Then I'd write 4 unittests:

extern_flag_not_passed_when_compiling_macro_2015
extern_flag_passed_when_compiling_macro_2018
extern_flag_not_passed_when_compiling_macro_wrapper_rule_2015
extern_flag_passed_when_compiling_macro_wrapper_rule_2015

Reply:
done